cont3xt/vueapp/src/components/pages/AuditHistory.vue

```diff
@@ -0,0 +1,273 @@
+        {
+          label: 'Indicator',
+          key: 'indicator',
+          sortable: true,
+          setWidth: '30rem'
+        },
+        {
+          label: 'Tags',
+          key: 'tags',
+          sortable: true
+        },
+        {
+          label: 'Options',
+          key: 'queryOptions',
+          sortable: true,
+          setWidth: '15rem'
+        },
+        {
+          label: 'Count',
+          key: 'resultCount',
+          sortable: true,
+          setWidth: '4rem',
+          tdClass: 'text-right',
+          formatter: this.orQuestionMark
+        },
+        {
+          label: 'Took',
+          key: 'took',
+          sortable: true,
+          setWidth: '4rem',
+          tdClass: 'text-right',
+          formatter: this.millisecondStr
+        }
+      ],
+      sortBy: 'issuedAt',
+      sortDesc: true,
+      filter: ''
+    };
+  },
+  watch: {
+    timeRangeInfo (newVal) {
+      // only re-search audits if there has already been a search, and the time-range has actually changed
+      if (this.lastTimeRangeInfoSearched != null &&
+          (newVal.startMs !== this.lastTimeRangeInfoSearched.startMs || newVal.stopMs !== this.lastTimeRangeInfoSearched.stopMs)) {
+        this.loadAuditsFromSearch();
+      }
+    },
+    filter () {
+      this.loadAuditsFromSearch();
+    }
+  },
+  methods: { /* page methods ---------------------------------------- */
+    clearSearchTerm () {
+      this.filter = '';
+    },
+    orQuestionMark (obj) {
+      return obj ?? '?';
+    },
+    millisecondStr (msNum) {
+      return msNum ? `${msNum}ms` : '?';
+    },
+    reissueSearchLink (log) {
+      const otherQueryParams = Object.entries(log.queryOptions).map(([key, value]) => `&${key}=${value}`).join('');
+      return `?b=${window.btoa(log.indicator)}${otherQueryParams}`;
+    },
+    deleteLog (id) {
+      AuditService.deleteAudit(id).then(() => {
+        this.auditLogs = this.auditLogs.filter(log => log._id !== id);
+      }).catch((err) => console.log('ERROR - ', err));
+    },
+    customFilter (data, filterBy) {
+      const lowerFilter = filterBy.toLowerCase();
+      const simpleFilter = (value) => {
+        return value.toLowerCase().includes(lowerFilter);
+      };
+      const arrayFilter = (arr) => {
+        return arr.some(el => simpleFilter(el));
+      };
+      return simpleFilter(data.iType) || simpleFilter(data.indicator) || arrayFilter(data.tags);
+    },
+    loadAuditsFromSearch () {
+      this.lastTimeRangeInfoSearched = JSON.parse(JSON.stringify(this.timeRangeInfo));
+      AuditService.getAudits({
+        startMs: this.timeRangeInfo.startMs,
+        stopMs: this.timeRangeInfo.stopMs,
+        searchTerm: this.filter === '' ? undefined : this.filter.toLowerCase()
+      }).then(audits => {
+        this.auditLogs = audits;
+      });
+    }
+  },
+  mounted () {
+    this.loadAuditsFromSearch();
+  }
+};
+</script>
+
+<style scoped>
+.indicator-limit-width {
+  max-width: 30rem;
```

Comment:
github puts this in red, something wrong with it?

Reply:
It looks like github highlights set-sizings (i.e. pixel or rem measurements)—but in this case that is what I meant since I want to avoid massive strings from blowing up the table. I would have preferred relative measurements like percentages, but I can't find a good way for them to play nice with set-sized features like the buttons and timestamps.